single_kernel_mongo/events/backups.py

```diff
@@ -207,6 +223,7 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent) -> None:  #
     def _on_s3_relation_broken(self, event: RelationBrokenEvent) -> None:
         """Proceed on s3 relation broken."""
         self.manager.cleanup_certs_and_restart(event.relation)
+        self.manager.state.statuses.clear(scope="unit", component=self.manager.name)
```

Comment:
What is the current behavior when a backup/restore is running?

Reply:
Excellent question !
I think it would leave a broken backup and maybe the restarts would fail.

I think it would be safer to delay that event until the backup/restore finishes!

I can add that logic in a loop so that we wait until the backup/restore finishes.

Should we handle it the same way the shard waits for draining ?
I think so because we can't defer that event and we really want to wait until the backup is finished.

Reply:
I would agree, if a backup / restore is running, we should protect the user from a broken state - so similar to shards draining handling it is

Reply:
"fun fact" we don't have any kind of logic for failed backups for now, what should I do ?

Reply:
Summary of offline conversation: we wait for any currently backup or restore to complete if any. In case of an error, we do not retry.